kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+
+int sem_destroy(sem_t *semaphore)
+{
+	k_sem_reset(semaphore->sem);
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	int count;
+
+	count = k_sem_count_get(semaphore->sem);
+	*value = count;
+
+	return 0;
+}
+
+int sem_init(sem_t *semaphore, int pshared, unsigned int value)
+{
+	if ((semaphore->sem->limit) == 0) {
+		return -1;
+	}
+	ARG_UNUSED(pshared);
+	__ASSERT((pshared == 1) || (pshared == 0), "not correct pshared");
+	k_sem_init(semaphore->sem, value, semaphore->sem->limit);
+
+	return 0;
+}
+
+int sem_post(sem_t *semaphore)
+{
+	k_sem_give(semaphore->sem);
+
+	if (semaphore == NULL) {
+		return -1;
+	} else {
```

Comment:
you could remove the else here. returning 0 will always happen if semaphore is not null.

Apply that comment everywhere, you are doing it in many places, it's useless.